### PR TITLE
add vendor properties for php sdk generation

### PR DESCRIPTION
**Pull request by jvc9109**
- Some `x-properties` have been added to schemas
- `BankAccountCreatePlain.yaml` refactor to be a discriminator map.
- Some schemas have been rename to avoid class name collision in php sdk generation

### CODE REVIEW

File: openapi/components/schemas/BankAccounts/BankAccountCreatePlain.yaml [hunk 1]
```diff
@@ -1,11 +1,18 @@
 title: Bank Account
-anyOf:
-  - $ref: ./AccountNumberTypes/BBANType.yaml
-  - $ref: ./AccountNumberTypes/IBANType.yaml
+type: object
+required: 
+  -  accountNumberType
+## Preference is without the discriminator even if it is more accurate. Discriminator needed for php sdk generation
```
Comment (justnero) [inline openapi/components/schemas/BankAccounts/BankAccountCreatePlain.yaml:5]:
```suggestion
## Discriminator is required for sdk generation
```

File: openapi/components/schemas/PaymentInstructions/PaymentToken.yaml [hunk 2]
```diff
@@ -6,3 +6,4 @@ properties:
   token:
     description: Payment Token ID.
     type: string
+x-abstract-parent: PaymentInstruction
```
Comment (justnero) [inline openapi/components/schemas/PaymentInstructions/PaymentToken.yaml:9]:
```suggestion
x-abstract-parent: PaymentInstruction

```

File: openapi/components/schemas/PaymentInstruction.yaml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 oneOf:
   - $ref: PaymentInstructions/PaymentToken.yaml
-  - $ref: PaymentInstructions/PaymentInstrument.yaml
+  - $ref: PaymentInstructions/PaymentInstructionInstrument.yaml
   - $ref: PaymentInstructions/PaymentMethods.yaml
   - $ref: PaymentCards/PaymentCardCreatePlain.yaml
   - $ref: BankAccounts/BankAccountCreatePlain.yaml
+x-no-generate: true
```
Comment (justnero) [inline openapi/components/schemas/PaymentInstruction.yaml:7]:
```suggestion
x-no-generate: true

```

Comment (justnero) [inline openapi/components/schemas/PaymentInstruction.yaml:7]: Please add a new line at the end of the file.

File: openapi/components/schemas/PaymentInstructions/PaymentToken.yaml (hunk 2 above)
Comment (justnero) [inline openapi/components/schemas/PaymentInstructions/PaymentToken.yaml:9]: Please add a new line at the end of the file.

File: openapi/components/schemas/PaymentInstruments/ValueObjects/PaymentInstrumentValueObject.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 type: object
 anyOf:
-  - $ref: ./VaultedInstrument.yaml
   - $ref: ./AlternativePaymentInstrument.yaml
+  - $ref: ./VaultedInstrument.yaml
```
Comment (justnero) [inline openapi/components/schemas/PaymentInstruments/ValueObjects/PaymentInstrumentValueObject.yaml:4]: Is it an intended change? And if so, what does it do?

File: openapi/components/schemas/ReadyToPay/Adjust/AdjustPaymentMethod.yaml [hunk 5]
```diff
@@ -12,8 +12,9 @@ discriminator:
 properties:
   paymentMethod:
     description: The payment method name.
-  feature:
+  feature:
```
Comment (justnero) [inline openapi/components/schemas/ReadyToPay/Adjust/AdjustPaymentMethod.yaml:15]:
```suggestion
  feature:
```

File: openapi/components/schemas/ReadyToPay/Adjust/AdjustPaymentMethod.yaml [hunk 6]
```diff
@@ -12,8 +12,9 @@ discriminator:
 properties:
   paymentMethod:
     description: The payment method name.
-  feature:
+  feature: 
     description: |
       The specific payment method feature like `Google Pay`, `Apple Pay` or `Plaid`.
 
       If set to `null`, then there are no features. This applies to the payment method.
+x-no-abstract-methods: true
```
Comment (justnero) [inline openapi/components/schemas/ReadyToPay/Adjust/AdjustPaymentMethod.yaml:20]: Please add a new line at the end of the file.

File: openapi/components/schemas/ReadyToPay/Adjust/AdjustReadyToPayGeneric.yaml [hunk 7]
```diff
@@ -11,4 +11,4 @@ allOf:
               enum:
                 - paypal
       feature:
-        nullable: true
+        nullable: true
```
Comment (justnero) [inline openapi/components/schemas/ReadyToPay/Adjust/AdjustReadyToPayGeneric.yaml:14]: Please add a new line at the end of the file.

File: openapi/components/schemas/ReadyToPay/Adjust/AdjustReadyToPayPaymentCard.yaml [hunk 8]
```diff
@@ -11,4 +11,4 @@ allOf:
         nullable: true
         allOf:
           - $ref: ../Features/ApplePayFeatureName.yaml
-          - $ref: ../Features/GooglePayFeatureName.yaml
+          - $ref: ../Features/GooglePayFeatureName.yaml
```
Comment (justnero) [inline openapi/components/schemas/ReadyToPay/Adjust/AdjustReadyToPayPaymentCard.yaml:14]: Please revert this change, trailing new line is better than no trailing new line.

File: openapi/components/schemas/ReadyToPay/Adjust/AdjustReadyToPayPaypal.yaml [hunk 9]
```diff
@@ -10,4 +10,4 @@ allOf:
       feature:
         nullable: true
         allOf:
-          - $ref: ../Features/PayPalBillingAgreementFeatureName.yaml
+          - $ref: ../Features/PayPalBillingAgreementFeatureName.yaml
```
Comment (justnero) [inline openapi/components/schemas/ReadyToPay/Adjust/AdjustReadyToPayPaypal.yaml:13]: Please revert this change, trailing new line is better than no trailing new line.

**Review by noon-dawg — COMMENTED** [review]
All small suggestions

**Review by justnero — DISMISSED** [review]

**Review by justnero — DISMISSED** [review]

**Review by noon-dawg — DISMISSED** [review]

**Review by justnero — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]
